### PR TITLE
Lims 3268

### PR DESCRIPTION
**Pull request by mayadafahim**

### CODE REVIEW

File: ui_testing/testcases/base_test.py [hunk 1]
```diff
@@ -65,6 +67,18 @@ def get_active_article_with_tst_plan(self, test_plan_status='complete'):
                 self.base_selenium.LOGGER.info(' + Archived.')
         else:
             return {}
+        
+    
+    def get_active_test_unit(self, search, material_type='Raw Material'):
```
Comment (0xIslamTaha) [inline ui_testing/testcases/base_test.py]:
```suggestion
    def get_active_test_unit_with_material_type(self, search, material_type='Raw Material'):
```

File: ui_testing/testcases/base_test.py [hunk 2]
```diff
@@ -65,6 +67,18 @@ def get_active_article_with_tst_plan(self, test_plan_status='complete'):
...
+        self.base_selenium.LOGGER.info(' + Get Test Unit with  type {} .'.format(search))
+        self.test_unit.get_test_units_page()
+        test_units = self.test_unit.search(search)
+        test_units_dict = [self.base_selenium.get_row_cells_dict_related_to_header(row=test_unit) for
+                                    test_unit in test_units[:-1]]
+        for test_unit_dict in test_units_dict:
+            if test_unit_dict['Type'] == search and test_unit_dict['Material Type'].find(material_type) != -1:
```
Comment (0xIslamTaha) [inline ui_testing/testcases/base_test.py]:
```suggestion
            if test_unit_dict['Type'] == search and material_type in test_unit_dict['Material Type']:
```

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 3]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+        """
+        self.base_selenium.LOGGER.info('Running test case to create an existing order with test units')
+        test_units_list = []
+        test_unit_dict = self.get_active_test_unit(search='Qualitative', material_type='All')
+        qualt_test_unit = test_unit_dict
+        if qualt_test_unit:
+            self.base_selenium.LOGGER.info('Retrieved test unit ' + test_unit_dict['Test Unit Name'])
+            test_units_list.append(qualt_test_unit['Test Unit Name'])
+        test_unit_dict = self.get_active_test_unit(search='Quantitative')
+        quan_test_unit = test_unit_dict
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: use `test_unit_dict` there is no need to create new var

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 4]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+        if quan_test_unit:
+            self.base_selenium.LOGGER.info('Retrieved test unit ' + test_unit_dict['Test Unit Name'])
+            test_units_list.append(quan_test_unit['Test Unit Name'])
+        test_unit_dict = self.get_active_test_unit(search='Quantitative Mibi')
+        quan_mibi_test_unit = test_unit_dict
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: use test_unit_dict

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 5]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+    def test021_create_existing_order_with_test_units(self):
+        """
+        New: Orders: Create an existing order with test units
+
+        LIMS-3268
+        """
+        self.base_selenium.LOGGER.info('Running test case to create an existing order with test units')
+        test_units_list = []
+        test_unit_dict = self.get_active_test_unit(search='Qualitative', material_type='All')
+        qualt_test_unit = test_unit_dict
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: use test_unit_dict

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 6]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+        latest_order_data = self.base_selenium.get_row_cells_dict_related_to_header(
+            row=orders_analyess[0])
+        self.assertEqual(
+           created_order.replace("'", ""), latest_order_data['Order No.'].replace("'", ""))
+        
+        self.analyses_page.open_child_table(source=orders_analyess[0])
+        rows_with_childtable = self.analyses_page.result_table(element='general:table_child')
+        success = 'true'
+        for row in rows_with_childtable[:-1]:
+
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: remove empty line

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 7]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+            row=orders_analyess[0])
+        self.assertEqual(
+           created_order.replace("'", ""), latest_order_data['Order No.'].replace("'", ""))
+        
+        self.analyses_page.open_child_table(source=orders_analyess[0])
+        rows_with_childtable = self.analyses_page.result_table(element='general:table_child')
+        success = 'true'
+        for row in rows_with_childtable[:-1]:
+            
+            rows_with_headers=self.base_selenium.get_row_cells_dict_related_to_header(row=row, table_element='general:table_child')
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]:
```suggestion
            row_with_headers=self.base_selenium.get_row_cells_dict_related_to_header(row=row, table_element='general:table_child')
```

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 8]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+            testunit_name = rows_with_headers['Test Unit']
+            self.base_selenium.LOGGER.info(testunit_name)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]:
```suggestion
            self.base_selenium.LOGGER.info(" + Test unit : {}".format(testunit_name))
```

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 9]
```diff
@@ -819,6 +819,108 @@ def test019_validate_order_no_archived_exists(self):
...
+            if testunit_name not in test_units_list:
+                success = 'false'
+        self.assertEqual(
+           'true', success)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: replace from 921:824 by
`self.assertIn(testunit_name, test_units_list)`

File: ui_testing/pages/order_page.py [hunk 10]
```diff
@@ -111,13 +111,56 @@ def create_new_order(self, material_type='', article='', contact='', test_plan='
...
+        self.set_contact(contact=contact)
+        order_no = self.get_no()
+        for test_unit in test_units:
+            self.set_test_unit(test_unit)
+        
+        self.save(save_btn='order:save_btn')
+        self.base_selenium.LOGGER.info(' + Order created with no : {} '.format(order_no))
+        return order_no
+    
+    def create_existing_order_with_multiple_test_units(self, no='', material_type='', article='', contact='', test_units='', multiple_suborders=0):
```
Comment (0xIslamTaha) [inline ui_testing/pages/order_page.py]:
```suggestion
    def create_existing_order_with_multiple_test_units(self, no='', material_type='', article='', contact='', test_units=[], multiple_suborders=0):
```

**Comment by mayadafahim** [comment]
closed and done in PR30

**Pull request closed without merging**